### PR TITLE
feat: Bare workflow support for Expo builds

### PR DESCRIPTION
**Pull request by ml019**
## Description
- Add support for bare workflow configurations.
- Add support for app extensions which have their own Info.plist files, bundle ids etc.
- Use the APP_REFERENCE, if present, to define the ShortVersionString. This needs to be set in app extensions to the same value as the app bundle.
- Ensure metro bundler not running before attempting a build, and stop it after the build is finished

## Motivation and Context
SDK37 deprecated the use of Expokit in favour of the bare workflow.

Use of push notifications using onesignal service introduced the need to support app extensions.

## How Has This Been Tested?
Customer build

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves the structure or operation of the implementation)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Followup Actions

- [x] None

## Checklist:

- [ ] My change requires a change to the [documentation](https://github.com/hamlet-io/docs).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] None of the above.

### CODE REVIEW

File: cli/runExpoAppPublish.sh [hunk 1]
```diff
@@ -306,10 +321,9 @@ function main() {
   # Update the app.json with build context information - Also ensure we always have a unique IOS build number
   # filter out the credentials used for the build process
   jq --slurpfile envConfig "${CONFIG_FILE}" \
-    --arg RELEASE_CHANNEL "${RELEASE_CHANNEL}" \
     --arg BUILD_REFERENCE "${BUILD_REFERENCE}" \
     --arg BUILD_NUMBER "${BUILD_NUMBER}" \
-    '.expo.releaseChannel=$RELEASE_CHANNEL | .expo.extra.build_reference=$BUILD_REFERENCE | .expo.ios.buildNumber=$BUILD_NUMBER | .expo.extra=.expo.extra + $envConfig[]["AppConfig"]' <  "./app.json" > "${tmpdir}/environment-app.json"
```
Comment (roleyfoley) [inline cli/runExpoAppPublish.sh:312]: Any reason why release channel was removed? It does warn about it but some of the apps we have deployed rely on it.

Comment (ml019) [reply cli/runExpoAppPublish.sh:312]: Only to remove the warning. If its not part of the app.json schema, then its not going to used for anything as far as Expo is concerned. How do apps end up using it?

Comment (roleyfoley) [reply cli/runExpoAppPublish.sh:312]: It's still required, expo inject it as part of their own build process and inject it just before they publish the OTA. This is why its not pat of the expected values from the user so they give you a warning about it

Comment (ml019) [reply cli/runExpoAppPublish.sh:312]: ok - I can revert that change. I need to add setting of the wait time for OTA as well as Luke tells me setting it to zero (as is currently the default) basically causes the app to bypass OTA checking.

Comment (ml019) [reply cli/runExpoAppPublish.sh:312]: I've reverted the code to include the release channel, and added the WaitMs parameter to trigger an OTA check on startup.

**Comment by ml019** [comment]
Ready for another review please @roleyfoley.

**Pull request merged**